### PR TITLE
Added global response handlers.

### PR DESCRIPTION
**Pull request by scobbe**

### CODE REVIEW

File: src/main/java/com/dropbox/core/DbxRequestUtil.java [hunk 1]
```diff
@@ -278,44 +278,63 @@ public static String parseErrorBody(String requestId, int statusCode, byte[] bod
...
                     requestId,
                     "unexpected HTTP status code: " + response.getStatusCode() + ": " + message,
                     response.getStatusCode()
                 );
         }
+
+        DbxGlobalCallbackFactory factory = DbxRequestUtil.sharedCallbackFactory;
+        if (factory != null) {
+            DbxNetworkErrorCallback callback = factory.createNetworkErrorCallback();
+            callback.setNetworkError(networkError);
```
Comment (northnose) [inline src/main/java/com/dropbox/core/DbxRequestUtil.java]: Does `DbxNetworkErrorCallback` need to be a `Runnable`? Instead of having `setNetworkError()` + `run()`, would it be simpler to just have one method like `onNetworkError(T error)`?

File: src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java [hunk 2]
```diff
@@ -0,0 +1,17 @@
...
+ * Client should implement methods in this class to allow for consistent, global handling of route-specific error types,
+ * as well as general network errors. Normally, error handling is done on a request-by-request basis.
+ * However, it is convenient to handle some error behavior consistently, regardless of the request or the endpoint.
+ * For implementing global error handling for general networking errors like an auth error, `createNetworkErrorCallback`
+ * should be implemented. For implementing global error handling for route-specific errors, the
+ * `createRouteErrorCallback` should be implemented.
+ */
+public interface DbxGlobalCallbackFactory {
+    // Should instantiate separate callback object on each method invocation
+    <T> DbxRouteErrorCallback<T> createRouteErrorCallback(T routeError);
```
Comment (northnose) [inline src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java]: I think you can overload these, i.e. have methods like 

```
DbxRouteErrorCallback<ListFolderError> createRouteErrorCallback(ListFolderError routeError);
DbxRouteErrorCallback<LookupError> createRouteErrorCallback(LookupError routeError);
...
```
It could also be an abstract class so implementors need only add the ones they care about.

Just suggesting an alternative that doesn't use a bunch of if/elses (if it even works? my Java is not that great), but feel free to keep it this way since this is fine too.

File: src/main/java/com/dropbox/core/DbxRequestUtil.java (hunk 1 above)
Comment (scobbe) [inline src/main/java/com/dropbox/core/DbxRequestUtil.java]: Good idea! I changed this.

File: src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java (hunk 2 above)
Comment (scobbe) [inline src/main/java/com/dropbox/core/v2/callbacks/DbxGlobalCallbackFactory.java]: This is good intuition, and what I originally tried to implement. However, Java strangely doesn't do dynamic overloaded method resolution at the parameter level, only the object level.

Read more [here](http://stackoverflow.com/questions/1572322/overloaded-method-selection-based-on-the-parameters-real-type).

tl;dr the parameter type only influences the overloaded method resolution at compile time, not run time. Given that the route error is generic at compile time, it always resolves to the declared generic method (not the more specific `ListFolderError` method).

Therefore, I've opted to keep the existing implementation, because I can't think of a better way of doing it.
